ol.animation.rotate does not take shortest path.

When using `ol.animation.rotate` it does not always take the shortest direction to the desired rotation. Whenever the shortest direction crosses `0` then it goes around the long way.

I wanted to make a Pull Request for this issue since it's easy to solve and I don't see any reason why the rotation shouldn't animate into the shortest direction by default.
But since the freshly checked out Openlayers throws errors after a make without changes I won't be able to make a pull request with changes either.
I'm totally new to the Openlayers Code. Did I forget something?
```bash
> make check-deps
Program git OK
Program node OK
Program python OK
Program java OK
```
```bash
> make check
Running eslint...
node tasks/build.js config/ol.json build/ol.js
info ol Parsing dependencies
ERR! Unsatisfied dependency "ol.ext.rbush" in script: /home/openlayers/src/ol/structs/rbush.js 
Makefile:271: die Regel für Ziel „build/ol.js“ scheiterte
make: *** [build/ol.js] Fehler 1
```

--- a/src/ol/view.js
+++ b/src/ol/view.js
@@ -15,6 +15,7 @@ goog.require('ol.extent');
 goog.require('ol.geom.GeometryType');
 goog.require('ol.geom.Polygon');
 goog.require('ol.geom.SimpleGeometry');
+goog.require('ol.math');
 goog.require('ol.obj');
 goog.require('ol.proj');
 goog.require('ol.proj.Units');
@@ -295,7 +296,10 @@ ol.View.prototype.animate = function(var_args) {
 
     if (options.rotation !== undefined) {
       animation.sourceRotation = rotation;
-      animation.targetRotation = options.rotation;
+      var delta =
+          ol.math.modulo(options.rotation - rotation + Math.PI, 2 * Math.PI) -
+          Math.PI;
+      animation.targetRotation = rotation + delta;
       rotation = animation.targetRotation;
     }
 

--- a/test/spec/ol/view.test.js
+++ b/test/spec/ol/view.test.js
@@ -494,6 +494,21 @@ describe('ol.View', function() {
       });
     });
 
+    it('takes the shortest angle to the target rotation', function(done) {
+      var view = new ol.View({
+        center: [0, 0],
+        zoom: 0,
+        rotation: Math.PI / 180 * 1
+      });
+      view.animate({
+        rotation: Math.PI / 180 * 359,
+        duration: 0
+      }, function() {
+        expect(view.getRotation()).to.roughlyEqual(Math.PI / 180 * -1, 1e-12);
+        done();
+      });
+    });
+
     it('calls a callback when animation completes', function(done) {
       var view = new ol.View({
         center: [0, 0],

